.github/workflows/sync-opencraft-forks-with-upstream.yml

```diff
@@ -0,0 +1,86 @@
+name: Sync OpenCraft Forks with upstream
+
+# Controls when the workflow will run
+on:
+  # TODO: Uncomment this out when its ready to be merged
+  # schedule:
+  #   - cron: '0 0 * * 1'  # Runs every Monday 00:00 UTC
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  sync:
+    strategy:
+      matrix:
+        include:
+          # The fork repo provided needs to be in the {owner}/{repo} format
+          - fork_repo: open-craft/edx-platform
+            fork_branch: opencraft-release/palm.1
+            upstream_repo: https://github.com/openedx/edx-platform
+            upstream_branch: open-release/palm.master
+
+    runs-on: ubuntu-latest
+    env:
+      # TODO: set this to the OpenCraft Requirements Bot PAT
+      GITHUB_TOKEN: ${{ secrets.GITHUB_BOT_TOKEN }}
+
+    steps:
+      - name: Get and store current datetime stamp
+        id: current-datetime-stamp
+        run: echo "DATE_TIME_STAMP=$(date +'%Y%m%d-%s')" >> "$GITHUB_OUTPUT"
+
+      - name: Construct and store sync branch name
+        id: sync-branch
+        run: echo "SYNC_BRANCH=sync-${{ matrix.upstream_branch }}-${{ steps.current-datetime-stamp.outputs.DATE_TIME_STAMP }}" >> "$GITHUB_OUTPUT"
+
+      - name: Configure git to use GitHub CLI for credentials
+        run: gh auth setup-git -h github.com
+
+      - name: Checkout Forked Repo
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ matrix.fork_repo }}
+          token: ${{ secrets.GITHUB_BOT_TOKEN }}
+
+      - name: Add Remote Upstream
+        run: git remote add upstream ${{ matrix.upstream_repo }}
+
+      - name: Fetch Latest from Upstream
+        run: git fetch upstream
+
+      - name: Create a new branch off Upstream master
+        run: git checkout -b ${{ steps.sync-branch.outputs.SYNC_BRANCH }} upstream/${{ matrix.upstream_branch }}
+
+      - name: Check for changes Upstream
+        id: check-sync-required
+        run: echo "SYNC_REQUIRED=$(git log ${{ steps.sync-branch.outputs.SYNC_BRANCH }} ^${{ matrix.fork_branch }} | grep -q . && echo true)" >> "$GITHUB_OUTPUT"
```

Comment:
We should also inform the user, e.g., by echoing "No new changes. Skipping the next steps.".